File: api/src/main/java/com/aiko/api/model/Line.java
package com.aiko.api.model;

import java.util.List;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.OneToMany;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;

import lombok.Data;

@Data
@Entity
public class Line {
  
  @Id
  @GeneratedValue(strategy=GenerationType.IDENTITY)
  private long id;


  private String name;

  @ManyToMany
  @JoinTable(
  name = "stop_line", 
  joinColumns = @JoinColumn(name = "line_id"), 
  inverseJoinColumns = @JoinColumn(name = "stop_id"))
  private List<Stop> stops;

  @OneToMany(mappedBy = "line")
  private List<Vehicle> vehicles;
  
}

File: TestBackEnd.sln/src/TestBackEnd.Domain/Entities/Line.cs
using System.Collections.Generic;

namespace TestBackEnd.Domain.Entities
{
    public class Line : Base
    {
        public string Name { get; private set; }

        public ICollection<Stop> Stops { get; set; }

        public ICollection<Vehicle> Vehicles { get; set; }

        public Line() { }

        public Line(string name, ICollection<Stop> stops, ICollection<Vehicle> vehicles)
        {
            Name = name;
            Stops = stops;
            Vehicles = vehicles;
        }
    }
}